frontend/web/components/pages/EnvironmentSettingsPage.js

```diff
@@ -61,7 +61,7 @@ const EnvironmentSettingsPage = class extends Component {
     const env = ProjectStore.getEnvs().find(
       (v) => v.api_key === this.props.match.params.environmentId,
     )
-    this.setState({ env })
+    this.setState({ env, description: env.description ?? "" })
```

Comment:
I'm interested to know why `description` has to be handled differently? Shouldn't it just be updated as part of the environment object? 🤔

Reply:
My approach here was to make sure we're consistent in how we're getting and updating the data from, previously we were mixing internal states with context one, which in my opinion is more confusing and prone to bugs. So, either using environment object or description internal states is fine.
```
  value={
    typeof this.state.description === 'string'
      ? this.state.description
      : env.description
  }
```